framework/src/Volo.Abp.EntityFrameworkCore/Volo/Abp/Domain/Repositories/EntityFrameworkCore/EfCoreRepository.cs

```diff
@@ -107,7 +108,7 @@ public EfCoreRepository(IDbContextProvider<TDbContext> dbContextProvider)
 
     public async override Task<TEntity> InsertAsync(TEntity entity, bool autoSave = false, CancellationToken cancellationToken = default)
     {
-        CheckReadOnly();
+        CheckChangeTracking();
```

Comment:
Inserting/updating/deleting an entity work even if you don't track changes for queried entities.

Reply:
But the entity returned by the method or parameters is `no tracking`.

Reply:
Where it is done? I didn't see any code part that detaches inserted/updated entities.

Reply:
There might be code like this:

```
using (var uow = uowManager.Begin())
{
    var author = await authorRepository.InsertAsync(new Author(Guid.NewGuid(), "Douglas"));

    // Entity still under tracking
    author.Name = "Douglas2";
    await uow.CompleteAsync();
}
```

Reply:
So, in this example, `author` object is being tracked, and the name change is auto-saved, right? But the warning message says "Your changes may not be saved". Isn't that misleading the developer?

Reply:
The code I share will not work if we disable entity tracking.
But developers may use such code, so I think we should output warnings when entity tracking is disabled.